File: ros-humble/node.py
import sys

import rclpy.node


class NodeBase(rclpy.node.Node):

    def __init__(self, name):
        super().__init__(name)
        # log methods
        self.info = self.get_logger().info
        self.warn = self.get_logger().warn
        self.error = self.get_logger().error
        self.debug = self.get_logger().debug


# ros2 pkg create zjros2 --build-type ament_python --dependencies rclpy
def main(name="demo", node_t=NodeBase):
    rclpy.init(args=sys.argv)
    node = node_t(name=name)
    node.info(f"Successful initialization.")
    rclpy.spin(node)
    rclpy.shutdown()

File: ros-humble/node.cpp
#include <rclcpp/rclcpp.hpp>


class Node : public rclcpp::Node {

public:
    typedef std::shared_ptr<Node> SharedPtr;

    Node(const std::string &name) : rclcpp::Node(name) {}
};


// ros2 pkg create zjros2 --dependencies rclcpp
int main(int argc, char **argv) {
  rclcpp::init(argc, argv);

  auto node = std::make_shared<Node>("demo");
  RCLCPP_INFO(node->get_logger(), "Successful initialization.");

  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}
